Formula/caddy.rb

```diff
@@ -52,15 +51,35 @@ def plist
   end
 
   test do
-    port = free_port
-    begin
-      io = IO.popen("#{bin}/caddy -port #{port}")
-      sleep 2
-    ensure
-      Process.kill("SIGINT", io.pid)
-      Process.wait(io.pid)
+    require "socket"
+
+    server1 = TCPServer.new(0)
+    port1 = server1.addr[1]
+    server2 = TCPServer.new(0)
+    port2 = server2.addr[1]
+    server1.close
+    server2.close
```

Comment:
use `free_port` instead

Reply:
Do you have an example code snippet for me? I've copied this from another formula and haven't seen `free_port` used anywhere so far.

Reply:
The code you removed uses `free_port`

Reply:
Oh, you are right! I developed the test from scratch as the old test didn't actually test anything (actually I developed the whole formula from scratch).

Can I request two free ports with this?

```rb
port1 = free_port
port2 = free_port
```

Reply:
Yes, that's all you need